packages/core/src/index.ts

```diff
@@ -28,9 +28,11 @@ class ImageService extends Service {
   }
 
   register(source: ImageSource) {
-    const index = this.sources.length
-    this.sources.push(source)
-    return this[Context.current].collect('booru', () => delete this.sources[index])
+    return this[Context.origin].effect(() => {
+      const index = this.sources.length
+      this.sources.push(source)
+      return { dispose: () => delete this.sources[index] }
```

Comment:
```suggestion
      this.sources.push(source)
      return { dispose: () => remove(this.sources, source) }
```

remove 从 koishi 导入即可。